---
title: RNA-seq gene counts to DE genes from compass
output: html_notebook
date: 08/11/2019
---


import of the risk groups from CoMMpass
```{r}

library(data.table)
library(tidyverse)
library(survival)
library(survminer)

# import <- fread("C:/Users/pillo/OneDrive/Desktop/complete_database_1q_13_CoMMpass.txt")
import <- fread("D:/analisi_in_corso/risk_1q_13/CoMMpass/complete_database_1q_13_CoMMpass.txt")


# define the MM risk label from commpass broad + focal CNAs
import$AMP_1q_genes_all <- with(import, 
                                    ifelse(`AMP_maj_focal_ANP32E`==1 | `AMP_maj_focal_MCL1` ==1 | `AMP_maj_focal_CKS1B`==1, 1,0 ))
    
    
table(import$`AMP_maj_broad_chr_1q`, import$AMP_1q_genes_all)
table(import$`DEL_maj_broad_chr_13q`, import$`DEL_maj_focal_RB1`)


import$MMrisk_1q_all <- ifelse( import$`AMP_maj_broad_chr_1q` == 1 | import$AMP_1q_genes_all ==1, 1, 0)
import$MMrisk_13_all <- ifelse( import$`DEL_maj_broad_chr_13q` ==1 | import$`DEL_maj_focal_RB1` ==1, 1, 0)

import$MMrisk_CLASSIFIER_ALL <- 3  - import$MMrisk_1q_all - import$MMrisk_13_all
import$MMrisk_CLASSIFIER_ALL <- dplyr::recode(as.character(import$MMrisk_CLASSIFIER_ALL), "1"="risk1","2"="risk2","3"="risk3")

table(import$MMrisk_CLASSIFIER_ALL)

    
# define the dataframe with group assignments    
compassGroup <- import %>% select(Study_Visit_iD,MMrisk_CLASSIFIER_ALL) 


```

***

prerequisites (install packages)
```{r}
# BiocManager::install("edgeR", update = F)
# BiocManager::install("Glimma", update = F)
```
***
# [0] loading libraries and set analysis parameters
***
```{r}
library(ggplot2)
library(dplyr)
library(data.table)
library(readr)
library(edgeR)
library(limma)
library(Glimma)
library(RColorBrewer)
library(biomaRt)
library(knitr)

NAME<-"CoMMpass_risk1_vs_risk3" # set name of analysis

# setwd("C:/Users/pillo/OneDrive/Desktop/deg//") # set working directory
# setwd("D:/analisi_in_corso/risk_1q_13/CoMMpass/DEG_CoMMpass/")

opts_knit$set(root.dir = "D:/analisi_in_corso/risk_1q_13/CoMMpass/DEG_CoMMpass/")

DIR<-getwd()

```

***
# [1] data import and prep
***
1.1 import GENE COUNTS RAW DATA from SALMON RNA-seq analysis (file downloaded from coMMPass) + GROUPS file
```{r}
# 1.1 ____import GENE COUNTS RAW DATA from SALMON RNA-seq analysis (file downloaded from coMMPass)_____

importCounts <- fread("D:/CoMMpass_data/IA14a//Expression_Estimates/MMRF_CoMMpass_IA14a_E74GTF_Salmon_Gene_Counts.txt.gz") # import gene counts data

# define the dataframe with group assignments
groupImport<- compassGroup

# groupImport<- filter(groupImport, !is.na( t.11.14.CCND1 ) ) # SET FILTER for NAs (translocations)!

```

# 1.2 REORDERING SAMPLE COLUMNS - CRITICAL STEP
```{r}
# 1.2 ____REORDERING SAMPLE COLUMNS - CRITICAL STEP_____

importCounts<-as.data.frame(importCounts)
head(names(importCounts))

ord<-order(names(importCounts))

importCounts<- importCounts[,ord] # reordering step!

head(names(importCounts))
```

***

# [2] Analysis variables creation

2.1  integration of ENSEMBL gene annotation with HGNC_SYMBOL, START, END info downloaded from bioMart 
```{r}

genelist<-importCounts$GENE_ID #create commpass gene list

library(biomaRt)
human <- useMart("ensembl", dataset="hsapiens_gene_ensembl")

gene_coords=getBM(attributes=c("hgnc_symbol","ensembl_gene_id", "start_position","end_position"),
                  filters="ensembl_gene_id",
                  values=genelist, #download the additional annotation for the compass gene list
                  mart=human)

#import "bioMart_genes_annot.RData" prepared file if no internet connection available 
# load(paste0(DIR,"/bioMart_genes_annot.RData")) #NOTA BENE!!!!!

#creation of length variable
gene_coords$length = gene_coords$end_position - gene_coords$start_position

#rename the gene header of the import file
names(importCounts)[1]<-"ensembl_gene_id"

#merge the import file with the new downloaded annotations
mergedInfoCounts<- left_join(gene_coords, importCounts, by = "ensembl_gene_id") # NB! FROM 57997 genes TO 52460 genes
```


2.2  Select only DIAGNOSIS samples from CoMMpass 
```{r}
onlyDiagnosis<- dplyr::select(mergedInfoCounts, contains("_1_BM")) 

samplenamesD<-names(onlyDiagnosis)

prepData<- onlyDiagnosis # CREATION of prepData object
```


2.2b [OPTIONAL] traslocation sample filtering step 
```{r}
# 2.2b___OPT___ Exclude samples with translocations involving cycline deregulations ________

importTransloc<- fread(choose.files()) # import translocation info file ("CoMMpass_Translocation_cyclin.csv" )

drops <- filter(importTransloc, Cyclin_T == "yes")[[1]] #define samples with translocation

prova<-prepData[, !(names(prepData) %in% drops)] #exclude translocated patients from prepData

#checks

sum(names(prova)%in% drops)
sum(names(prova)%in%names(prepData)==F)

samplenamesDnoT<-names(prova)

prepData<-prova # !!! SAVE prepData !!!
```



2.2c [OPTIONAL] others sample type filtering steps
```{r}
#define the condition for keeping samples 
keep<- dplyr::filter(groupImport, groupImport$`1q` == 0) %>% dplyr::select(sample) %>% unlist()

prepData<- prepData[, names(prepData) %in% keep]
```




2.3 define GROUP variable in order to perform differential expression analysis between groups
```{r}
groupSelect<- dplyr::filter(groupImport, Study_Visit_iD %in% names(prepData)) # select the name list to filter


samplewithgroup<- groupSelect$Study_Visit_iD
## groupCheck<-filter(groupImport, sample %in% samplewithgroup)

#excluding samples without a group
onlyGroup<- dplyr::select(prepData, samplewithgroup)

prepData<-cbind(mergedInfoCounts[,c(1,2,5)],onlyGroup) # !!! SAVE prepData !!! + add 3 gene annotations cols 

dim(prepData)# row= genes, cols= info + samples

```

2.4  create essential variable necessary for DEG analysis 
```{r}
#_______essential 1: group
group<-as.factor(groupSelect$MMrisk_CLASSIFIER_ALL) 

length(group)
table(group)

#_______essential 2: counts
counts<-as.matrix(prepData[,c(4:ncol(prepData))])
dim(counts)
 
#_______essential 3: samples
samples<-data.frame(samples=names(prepData[-c(1:3)]))

#_______essential 4: genes
genes<-data.frame(genes=prepData[,1:3])

# analyze if there are duplicated genes
notdup<-genes[!duplicated(genes$genes.ensembl_gene_id),]
dup<-genes[duplicated(genes$genes.ensembl_gene_id),]
```

***
# [3] edgeR: RNA-seq anlysis normalization & preparation
***
3.1  data import in edgeR object
```{r}

library(edgeR)

x<- DGEList(counts,samples=samples,genes=genes)
dim(x)

# group assignment
x$samples$group<- group
```

3.2 data pre-processing 
```{r}
# trasformations from the raw scale to Counts Per Million (CPM) and logCMP
cpm<-cpm(x)
lcpm<-cpm(x, log=T)

#notice differences after trasnformation
# View(cpm[1:50,1:20])
# View(lcpm[1:50,1:20])
# View(counts[1:50,1:20])
```

**PLOT> grid of low-expr filter parameters**
```{r}
#~~~~~~~~~ OPTIONAL: plot a GRID to best choose the optimum parameters to filter lowexpr genes ~~~~~~~~~
# 
i_cpm<-vector()
j_pat<-vector()
c_keep<-vector()
for (i in seq(0.1,2,by=0.05)){   # try different CPM thresholds
  for (j in seq(1,30,by=1)){     # try different min patients cutoffs
    vec <- rowSums(cpm>i)>=j
    c<-table(vec)[2]             # use the kept genes as the result value
    c_keep<-append(c_keep,c)
    i_cpm<-append(i_cpm,i)
    j_pat<-append(j_pat,j)
  }
}

plot(c_keep)

#creation of a dataframe
data <- data.frame(i_cpm, j_pat, c_keep)

#plot with ggplot
library(ggplot2)
ggplot(data, aes(i_cpm, j_pat, z= c_keep)) +
  geom_tile(aes(fill = c_keep)) +
  theme_bw()

#plot with lattice
library("lattice")
par(mar=c(3,4,2,2))
levelplot(c_keep ~ i_cpm*j_pat, data=data  , xlab="cpm" , col.regions = heat.colors(100)[length(heat.colors(100)):1]   , main="")

#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

3.3 removing low expressed genes
```{r}
# see how many genes have 0 counts in every sample 
table(rowSums(x$counts==0)== ncol(x)) # ncol(x) = total number of samples 

# filter genes basing on low-expr thresholds
keep.exprs<- rowSums(cpm>1)>=5 # select the genes that have a cpm >1 (typical CPM threshold) in at least 5 samples (minimum significative group size)
table(keep.exprs)

x<-x[keep.exprs,, keep.lib.sizes=FALSE]
dim(x)
```

**PLOT> filtering results**
```{r}
# plotting the results of filtering

library(RColorBrewer)
nsamples <- ncol(x)
col <- brewer.pal(nsamples, "Paired")
par(mfrow=c(1,2))
plot(density(lcpm[,1]), col=col[1], lwd=2, ylim=c(0,0.21), las=2,
     main="", xlab="")
title(main="A. Raw data", xlab="Log-cpm")
abline(v=0, lty=3)
for (i in 2:nsamples){
  den <- density(lcpm[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
}
lcpm <- cpm(x, log=TRUE)
plot(density(lcpm[,1]), col=col[1], lwd=2, ylim=c(0,0.21), las=2,
     main="", xlab="")
title(main="B. Filtered data", xlab="Log-cpm")
abline(v=0, lty=3)
for (i in 2:nsamples){
  den <- density(lcpm[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
}
```

3.4 Normalization across samples in the experiment 
```{r}
x<- calcNormFactors(x, method = "TMM")
head(x$samples$norm.factors, n=20)
```

**PLOT> intersamples expr normalization boxplots**
```{r}
# explore the effect of normalization (duplicated and modified data to enhance the visual effect)

x2<-x[,1:50] # ONLY first 20 samples in this example
x2$samples$norm.factors <- 1
# x2$counts[,1]<- ceiling(x2$counts[,1]*0.05) # first sample counts are reduced to 5% of original values
# x2$counts[,2]<-x2$counts[,2]*5 # second sample counts are inflated to 500% of original value

library(RColorBrewer)
nsamples <- ncol(x)
col <- brewer.pal(nsamples, "Paired")


par(mfrow=c(1,2))
lcpm <- cpm(x2, log=TRUE)
boxplot(lcpm, las=2, col=col, main="")
title(main="A. Example: Unnormalised data",ylab="Log-cpm")

x2 <- calcNormFactors(x2)
x2$samples$norm.factors
lcpm <- cpm(x2, log=TRUE)
boxplot(lcpm, las=2, col=col, main="")
title(main="B. Example: Normalised data",ylab="Log-cpm")
```

***
# [4] limma & Glimma: identify DEG and visualize results 
***

4.1 Multi Dimensional Scaling (MDS) analysis plot (~PCA of the groups)
```{r}
library(limma)

# (search initial global evidence for a differential expression between groups)

lcpm <- cpm(x, log=TRUE)
```

**PLOT> MDS plot with limma**
```{r}
# MDS PLOT
par(mfrow=c(1,2))
col.group <- group
levels(col.group) <- brewer.pal(nlevels(col.group), "Set1")
col.group <- as.character(col.group)

plotMDS(lcpm, #labels=group, 
        pch=20,
        col=col.group, 
        dim=c(1,2))# plot dimension 1 vs dimension 2

title(main="A. Sample groups")

plotMDS(lcpm, #labels=group, 
        pch=20,
        col=col.group, 
        dim=c(2,3)) # plot dimension 2 vs dimension 3
```

**PLOT> MDS INTERACTIVE plot with Glimma**
```{r}
# Glimma interactive MDS viz - ALL dimensions
library(Glimma)
glMDSPlot(lcpm, labels="group", groups=x$samples$group,
          launch=TRUE) #launch = TRUE opens the plot in a browser
```

4.2 Differential expression experimental settings: DESIGN and CONTRAST matrices 
```{r}
#create a DESIGN MATRIX from group

design<- model.matrix(~0+group)
colnames(design)<-gsub("group","",colnames(design))

design[1:10,]

#create a CONTRAST MATRIX with comparison of interest between groups

# SET the groups

contr.matrix<- makeContrasts(
  r1v3 = `risk1`-`risk3`,
  r1v2 = `risk1`-`risk2`,
  r2vr3 = `risk2`-`risk3`,
  
  
  levels = colnames(design))
contr.matrix
```

4.3 VOOM function: removing HETEROSCEDASCITY from count data (and transform count data in logCPM) 
```{r}
v<- voom(x, design, plot=TRUE)
v[1:10,1:10]

# View(v$targets)
```

4.4 === Fitting linear models and bayesian t-test for comparisons of interest ====
```{r}

vfit<- lmFit(v,design) # calculate the mean expression level for each gene in the different groups

vfit<- contrasts.fit(vfit, contrasts = contr.matrix) # compare the groups

efit<- eBayes(vfit) # calculate statistical bayesian-variance-adjusted T-test to find significative differential expressed genes (pval<0.05)
```

**PLOT> voom plot: median-variance heteroscedascity correction**
```{r}
plotSA(efit)
```

**RESULTS> DE genes**
```{r}
# Examining the number of DE genes: Significance is defined using an adjusted p-value cutoff that is set at 5%
summary(decideTests(efit))

tab= topTable(efit, coef = 1, number=1000, adjust.method = "BH")
# View(tab)

topGenes<- tab[tab[,"adj.P.Val"]< 0.001,]
```

**PLOT> volcano plot of DEG (p<0.05)**
```{r}
volcanoplot(efit,coef = 1, highlight = 500, names=efit$genes$genes.hgnc_symbol)
title(main="Volcano plot of top 500 deregulated genes")
```

4.4b Strictier definition of significance 
```{r}
# "treat" method to calculate p-values from empirical Bayes moderated t-statistics with a minimum log-FC requirement. 

tfit <- treat(vfit, lfc=1) # lfc=Log2FoldChange (lfc=1 is then equal to 2 fold-change)
```

**RESULTS> strict DE genes**
```{r}

#Examining the number of strict DE genes (with minimum fold-change > 2)
dt <- decideTests(tfit)
summary(dt)

strictNum <- length(which(dt[,1]!=0))

tabStrict<- topTreat(tfit, coef=1, sort.by = "p", number=strictNum) # SET the correct number of strict DEG genes !
# View(tabStrict)
```

**PLOT> volcano plot of STRICT DEG**
```{r}
volcanoplot(tfit,coef = 1, highlight = strictNum, names= tfit$genes$genes.hgnc_symbol) # SET the correct number of strict DEG genes
title(main="Volcano plot of STRICT (fold-change>2) deregulated genes")
```

4.5 EXTRACT both upregolated and downregolated STRICT singificative DEG 

**RESULTS> gene LISTS**
```{r}
# common up and down DEG
de.common<- which(dt[,1]!=0)
length(de.common)

tfit$genes$genes.hgnc_symbol[de.common]

# up-regulated genes
de.up<- which(dt[,1]==1)
length(de.up)

tfit$genes$genes.hgnc_symbol[de.up]

# down-regulated genes
de.down<- which(dt[,1]== -1)
length(de.down)

tfit$genes$genes.hgnc_symbol[de.down]


# Examining individual DE genes from top to bottom
DEGresults<- topTreat(tfit, coef=1, n=Inf)
# View(DEGresults)
```

4.6  EXPORT the strict DE gene list and the analysis results 
```{r}
write.fit(tfit, dt, file=paste0("DEG_results_",NAME,".txt")) # analysis result
write_tsv(tabStrict,paste0("DEG_genes_",NAME,".txt")) # strict DE list
```

4.7 Useful graphical representations of differential expression results 
**PLOT> MD plot of STRICT DEG by limma**
```{r}
# MD plot with UPREGOLATED and DOWNREGOLATED strict significative genes
plotMD(tfit, column=1, status=dt[,1], main=colnames(tfit)[1], xlim=c(-8,13))
```

**PLOT> MD plot of STRICT DEG by Glimma**
```{r}
# Glimma interactive MD plot 
library(Glimma)
glMDPlot(tfit, coef=1, status=dt, main=colnames(tfit)[1],
         side.main="genes.hgnc_symbol", counts=x$counts, groups=group, launch=TRUE) # Glimma INTERACTIVE
```

**PLOT> HEATMAP**
```{r}
# HEATMAP of DEG

vplot<-v

# vplot<- vplot[,20:420] #reduce the plot size over a smaller subset of samples

vplot1.3<-vplot[,vplot$targets$group!="risk2"] #exlude group R2 from plot (if present)
table(vplot1.3$targets$group)
groupPlot<-as.character(vplot1.3$targets$group) 


groupPlot<-as.character(vplot1.3$targets$group)

groupColors<- gsub("risk3","green3",groupPlot)
groupColors<- gsub("risk1","orange",groupColors)

library(gplots)

DEGresults.topgenes <- DEGresults$genes.ensembl_gene_id[1:length(de.common)]

i <- which(vplot1.3$genes$genes.ensembl_gene_id %in% DEGresults.topgenes)

mycol <- colorpanel(1000,"blue","white","red")


heatmap.2(vplot1.3$E[i,], 
          scale = "row",    # NB. SCALED expr values! 
          labRow=vplot$genes$genes.hgnc_symbol[i],
          labCol = groupPlot,
          col= mycol,
          # hclustfun = function(x) hclust(x, method="ward.D2"),
          trace="none",
          density.info="none",
          dendrogram="column",
          ColSideColors = groupColors)
```


**PLOT> HEATMAP all genes**
```{r}
# HEATMAP of DEG

vplot<-v

groupPlot<-as.character(vplot$targets$group)

groupColors<- gsub("risk2","green3",groupPlot)
groupColors<- gsub("risk1","orange",groupColors)
groupColors<- gsub("risk3","royalblue1",groupColors)

library(gplots)

DEGresults.topgenes <- DEGresults$genes.ensembl_gene_id[1:length(de.common)]



CCgenes=getBM(attributes=c("hgnc_symbol","ensembl_gene_id"), 
                  filters="go_parent_term", 
                  values="GO:0007049", #download the additional annotation for the compass gene list
                  mart=human)


CellCycle.genes <- v$genes$genes.ensembl_gene_id[v$genes$genes.ensembl_gene_id %in% CCgenes$ensembl_gene_id]

i <- which(v$genes$genes.ensembl_gene_id %in% CCgenes$ensembl_gene_id)

mycol <- colorpanel(1000,"blue","white","red")


heatmap.2(vplot$E[i,], 
          scale = "row",    # NB. SCALED expr values! 
          labRow=vplot$genes$genes.hgnc_symbol[i],
          labCol = groupPlot,
          col= mycol,
          # hclustfun = function(x) hclust(x, method="ward.D2"),
          trace="none",
          density.info="none",
          dendrogram="column",
          ColSideColors = groupColors)
```

***
# [5] pathway/enrichment R analysis
***

5.1  Gene set testing with camera 
```{r}
load(url("http://bioinf.wehi.edu.au/software/MSigDB/mouse_c2_v5p1.rdata"))
idx <- ids2indices(Mm.c2,id=rownames(v))
cam.results <- camera(v,idx,design,contrast=contr.matrix[,1])
head(cam.results,10)


barcodeplot(efit$t[,1], index=idx$WENG_POR_TARGETS_GLOBAL_DN ,
            index2=idx$BYSTRYKH_HEMATOPOIESIS_STEM_CELL_RUNX1, main="title")
```


5.2  mroast  TO DO!
```{r}
# mroast()
```


```{r}
# heat map of deg for CCND1-VS-CCND2
vplot<-v


groupPlot<-as.character(vplot$targets$group)

groupColors<- gsub("risk2","green3",groupPlot)
groupColors<- gsub("risk1","orange",groupColors)
groupColors<- gsub("risk3","royalblue1",groupColors)

library(gplots)

DEGresults.topgenes <- DEGresults$genes.ensembl_gene_id[1:length(de.common)]

i <- which(vplot$genes$genes.hgnc_symbol %in% c("CCND1","CCND2"))

mycol <- colorpanel(1000,"blue","white","red")


heatmap.2(vplot$E[i,], 
          scale = "row",    # NB. SCALED expr values! 
          labRow=vplot$genes$genes.hgnc_symbol[i],
          labCol = groupPlot,
          col= mycol,
          # hclustfun = function(x) hclust(x, method="ward.D2"),
          trace="none",
          density.info="none",
          dendrogram="column",
          ColSideColors = groupColors)
```



**PLOT> HEATMAP all genes - CATEGORIES RISK2a RISK2b**
```{r}
# HEATMAP of DEG

vplot<-v


as.character(vplot$targets$group) %>% length()
groupSelect$MMrisk_CLASSIFIER_ALL %>% length()

(as.character(vplot$targets$group) == groupSelect$MMrisk_CLASSIFIER_ALL) %>% table

data.group <- import %>% filter(Study_Visit_iD %in% groupSelect$Study_Visit_iD)

# data.group$Study_Visit_iD == groupSelect$Study_Visit_iD

(data.group$MMrisk_CLASSIFIER_ALL == as.character(vplot$targets$group)) %>% table

data.group$MMrisk_CLASSIFIER_ALL_ab <- ifelse(data.group$MMrisk_CLASSIFIER_ALL=="risk2" & data.group$MMrisk_1q_all==1, 
                                              "risk2a",
                                              ifelse(data.group$MMrisk_CLASSIFIER_ALL=="risk2" & data.group$MMrisk_13_all ==1, 
                                                     "risk2b", 
                                                     data.group$MMrisk_CLASSIFIER_ALL))




groupPlot<- data.group$MMrisk_CLASSIFIER_ALL_ab

groupColors<- gsub("risk2a","red",groupPlot)
groupColors<- gsub("risk2b","green3",groupColors)
groupColors<- gsub("risk1","orange",groupColors)
groupColors<- gsub("risk3","royalblue1",groupColors)


library(gplots)

DEGresults.topgenes <- DEGresults$genes.ensembl_gene_id[1:length(de.common)]

i <- which(vplot$genes$genes.hgnc_symbol %in% c("CCND1","CCND2"))

mycol <- colorpanel(1000,"blue","white","red")


heatmap.2(vplot$E[i,], 
          scale = "row",    # NB. SCALED expr values! 
          labRow=vplot$genes$genes.hgnc_symbol[i],
          labCol = groupPlot,
          col= mycol,
          # hclustfun = function(x) hclust(x, method="ward.D2"),
          trace="none",
          density.info="none",
          dendrogram="column",
          ColSideColors = groupColors)

```


clustering su gruppo 2 da solo


```{r}
# HEATMAP of DEG

vplot<-v


vplot2<-vplot[,vplot$targets$group=="risk2"] #exlude group R2 from plot (if present)
table(vplot2$targets$group)
groupPlot<-as.character(vplot2$targets$group) 




as.character(vplot2$targets$group) %>% length()
groupSelect$MMrisk_CLASSIFIER_ALL %>% length()

(as.character(vplot2$targets$group) == groupSelect$MMrisk_CLASSIFIER_ALL) %>% table

data.group <- import %>% filter(Study_Visit_iD %in% groupSelect$Study_Visit_iD, MMrisk_CLASSIFIER_ALL=="risk2")

# data.group$Study_Visit_iD == groupSelect$Study_Visit_iD

(data.group$MMrisk_CLASSIFIER_ALL == as.character(vplot2$targets$group)) %>% table

data.group$MMrisk_CLASSIFIER_ALL_ab <- ifelse(data.group$MMrisk_CLASSIFIER_ALL=="risk2" & data.group$MMrisk_1q_all==1, 
                                              "risk2a",
                                              ifelse(data.group$MMrisk_CLASSIFIER_ALL=="risk2" & data.group$MMrisk_13_all ==1, 
                                                     "risk2b", 
                                                     data.group$MMrisk_CLASSIFIER_ALL))




groupPlot<- data.group$MMrisk_CLASSIFIER_ALL_ab

groupColors<- gsub("risk2a","red",groupPlot)
groupColors<- gsub("risk2b","green3",groupColors)
groupColors<- gsub("risk1","orange",groupColors)
groupColors<- gsub("risk3","royalblue1",groupColors)


library(gplots)

DEGresults.topgenes <- DEGresults$genes.ensembl_gene_id[1:length(de.common)]

i <- which(vplot$genes$genes.hgnc_symbol %in% c("CCND1","CCND2"))

mycol <- colorpanel(1000,"blue","white","red")


heatmap.2(vplot2$E[i,], 
          scale = "row",    # NB. SCALED expr values! 
          labRow=vplot$genes$genes.hgnc_symbol[i],
          labCol = groupPlot,
          col= mycol,
          # hclustfun = function(x) hclust(x, method="ward.D2"),
          trace="none",
          density.info="none",
          dendrogram="column",
          ColSideColors = groupColors)

```

clustering su gruppo 1 + gruppo 3

```{r}
# HEATMAP of DEG

vplot<-v


vplot1.3<-vplot[,vplot1.3$targets$group!="risk2"] #exlude group R2 from plot (if present)
table(vplot1.3$targets$group)
groupPlot<-as.character(vplot1.3$targets$group) 




as.character(vplot1.3$targets$group) %>% length()
groupSelect$MMrisk_CLASSIFIER_ALL %>% length()

(as.character(vplot1.3$targets$group) == groupSelect$MMrisk_CLASSIFIER_ALL) %>% table

data.group <- import %>% filter(Study_Visit_iD %in% groupSelect$Study_Visit_iD, MMrisk_CLASSIFIER_ALL!="risk2")

# data.group$Study_Visit_iD == groupSelect$Study_Visit_iD

(data.group$MMrisk_CLASSIFIER_ALL == as.character(vplot1.3$targets$group)) %>% table

data.group$MMrisk_CLASSIFIER_ALL_ab <- ifelse(data.group$MMrisk_CLASSIFIER_ALL=="risk2" & data.group$MMrisk_1q_all==1, 
                                              "risk2a",
                                              ifelse(data.group$MMrisk_CLASSIFIER_ALL=="risk2" & data.group$MMrisk_13_all ==1, 
                                                     "risk2b", 
                                                     data.group$MMrisk_CLASSIFIER_ALL))




groupPlot<- data.group$MMrisk_CLASSIFIER_ALL_ab

groupColors<- gsub("risk2a","red",groupPlot)
groupColors<- gsub("risk2b","green3",groupColors)
groupColors<- gsub("risk1","orange",groupColors)
groupColors<- gsub("risk3","royalblue1",groupColors)


library(gplots)

DEGresults.topgenes <- DEGresults$genes.ensembl_gene_id[1:length(de.common)]

i <- which(vplot$genes$genes.hgnc_symbol %in% c("CCND1","CCND2"))

mycol <- colorpanel(1000,"blue","white","red")


heatmap.2(vplot1.3$E[i,], 
          scale = "row",    # NB. SCALED expr values! 
          labRow=vplot$genes$genes.hgnc_symbol[i],
          labCol = groupPlot,
          col= mycol,
          # hclustfun = function(x) hclust(x, method="ward.D2"),
          trace="none",
          density.info="none",
          dendrogram="column",
          ColSideColors = groupColors)

```
