---
title: "1q&13 Clinical Analysis REPORT"
date: "03/12/2020"
output:
  html_document:
    df_print: tibble
  word_document: default
---

```{r include=FALSE}
library(data.table) 
library(tidyverse) 
library(survival) 
library(survminer) 
library(RODBC)
library(gmodels)
library(DT)

db_clin <- RODBC::odbcConnectAccess2007("C:/Users/mm_gr/Alma Mater Studiorum Università di Bologna/MM group - Documenti/EVERGREEN_inter-projects_files/MM_db_clinical_records.accdb")

clin <- sqlFetch(db_clin, "Extraction_1q13_301120", dec=",", na.strings="nv" )

db_exp<- RODBC::odbcConnectAccess2007("C:/Users/mm_gr/Alma Mater Studiorum Università di Bologna/MM group - Documenti/EVERGREEN_inter-projects_files/MM_db_experiments.accdb")
calls <- sqlFetch(db_exp, "SNP_calls_1q_13_181019", dec=",")

import <- inner_join(clin, calls, by=c("SNP_ARRAY_ESORDIO"="SNP"))

totPts <- nrow(import)


#______ MM RISK CREATION _____
import$AMP_1q_genes_all <- with(import, 
                                ifelse(`AMP_maj-focal_ANP32E`==1 | `AMP_maj-focal_MCL1` ==1 | `AMP_maj-focal_CKS1B`==1, 1,0 ))

import$MMrisk_1q_all <- ifelse( import$`AMP_maj-broad_chr_1q` == 1 | import$AMP_1q_genes_all ==1, 1, 0)
import$MMrisk_13_all <- ifelse( import$`DEL_maj-broad_chr_13q` ==1 | import$`DEL_maj-focal_RB1` ==1, 1, 0)

import$MMrisk_CLASSIFIER_ALL <- 3  - import$MMrisk_1q_all - import$MMrisk_13_all

import$MMrisk1 <- ifelse(import$MMrisk_CLASSIFIER_ALL==1, 1, 0)
import$MMrisk2 <- ifelse(import$MMrisk_CLASSIFIER_ALL==2, 1, 0)
import$MMrisk3 <- ifelse(import$MMrisk_CLASSIFIER_ALL==3, 1, 0)

import$MMrisk_AB_ALL <- ifelse(import$MMrisk_CLASSIFIER_ALL == 2 & import$MMrisk_1q_all ==1,
                               "2_1q", 
                               ifelse(import$MMrisk_CLASSIFIER_ALL == 2 & import$MMrisk_13_all ==1,
                                      "2_13",
                                      import$MMrisk_CLASSIFIER_ALL)
)

import$risk1_VS_1q_VS_other <- recode(import$MMrisk_AB_ALL, "1"="1q&13", "2_1q"="1q_only", "2_13"="other", "3"="other")



#______ ULTRA MM RISK CREATION _____

import$ULTRA_MMRISK <- ifelse(import$MMrisk_CLASSIFIER_ALL==1 & import$ISS==3 , "Ultra_High_Risk", 
                              ifelse(import$MMrisk_CLASSIFIER_ALL==3 & import$ISS==1, "Ultra_Low_Risk", "other"))



import$Ultra_High <- ifelse(import$ULTRA_MMRISK=="Ultra_High_Risk", 1, 0)
import$Ultra_Low <- ifelse(import$ULTRA_MMRISK=="Ultra_Low_Risk", 1, 0)


import$ULTRA_MMRISK %>% table



```

------------------------------------------------------------------------

------------------------------------------------------------------------

## Clinical data explorative analysis

#### **Protocol table**

```{r echo=FALSE}
import %>% group_by(PROTOCOL) %>% summarise(n=n(), perc=paste0(((n()/totPts)*100) %>% round(2),"%")) %>% datatable()
```

#### **Protocol-rev table**

rev: pts different from BO2005/EMN02 = AMB

```{r echo=FALSE}

import$PROTOCOL_REV <- recode(import$PROTOCOL, "FP-EMN02"="AMB", "FP - BO2005"="AMB", "DARA"="AMB", "FORTE"="AMB")

import %>% group_by(PROTOCOL_REV) %>% summarise(n=n(), perc=paste0(((n()/totPts)*100) %>% round(2),"%"))%>% datatable()

```

------------------------------------------------------------------------

#### Ig Isotpye-rev

rev: "IgA+BJ"="IgA"

```{r echo=FALSE, warning=FALSE}

import$IG_ISOTYPE_REV <- recode(import$IG_ISOTYPE, "IgA+BJ"="IgA", "IgA/BJ"="IgA")

import %>% group_by(IG_ISOTYPE_REV) %>% summarise(n=n(), perc=paste0(((n()/totPts)*100) %>% round(2),"%"))%>% datatable()

```

------------------------------------------------------------------------

#### Number Transplant Front-line

```{r echo=FALSE, warning=FALSE}

import %>% group_by(NUMBERS_TX_FRONT_LINE) %>% summarise(n=n(), perc=paste0(((n()/totPts)*100) %>% round(2),"%"))%>% datatable()
```

------------------------------------------------------------------------

#### Age

```{r echo=FALSE, message=FALSE, warning=FALSE}

import %>% arrange(AGE) %>%  ggplot(aes(1:totPts, AGE)) + geom_point(shape=1) + geom_hline(yintercept = 65, colour="blue")

import %>% group_by(AGE>65) %>% summarise(n=n(), perc=paste0(((n()/totPts)*100) %>% round(2),"%"))%>% datatable()
```

------------------------------------------------------------------------

------------------------------------------------------------------------

## 1q&13 variables Explorative analysis

#### 1q&13 classes

```{r echo=FALSE, warning=FALSE}

import %>% group_by(MMrisk_CLASSIFIER_ALL) %>% summarise(n=n(), perc=paste0(((n()/totPts)*100) %>% round(2),"%"))%>% datatable()
```

#### Amp 1q definition (Broad vs Genes+Broad)

```{r echo=FALSE, warning=FALSE}

import %>% group_by(`AMP_maj-broad_chr_1q`, AMP_1q_genes_all) %>% summarise(n=n(), perc=paste0(((n()/totPts)*100) %>% round(2),"%"))%>% datatable()
```

#### Del 13 definition (Broad vs Genes+Broad)

```{r echo=FALSE, warning=FALSE}
import %>% group_by(`DEL_maj-broad_chr_13q`, `DEL_maj-focal_RB1`) %>% summarise(n=n(), perc=paste0(((n()/totPts)*100) %>% round(2),"%"))%>% datatable()

```

#### divided group 2 1q&13 classes

```{r echo=FALSE, warning=FALSE}
import %>% group_by(MMrisk_AB_ALL) %>% summarise(n=n(), perc=paste0(((n()/totPts)*100) %>% round(2),"%"))%>% datatable()

```

------------------------------------------------------------------------

------------------------------------------------------------------------

## Survival analysis

```{r echo=FALSE, warning=FALSE}
OS <- Surv(import$OS_MONTHS, import$OS_EVENT)
PFS <- Surv(import$PFS_I_MONTHS, import$PFS_I_EVENT)
```

### Protocol - pval \*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}

ggsurvplot(survfit(OS ~ import$PROTOCOL_REV, data = import), pval = T, risk.table = T) + xlab("OS")
ggsurvplot(survfit(PFS ~ import$PROTOCOL_REV, data = import), pval = T, risk.table = T) +
xlab("PFS")
```

------------------------------------------------------------------------

### Old - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$OLD_0_1, data = import), pval = T, risk.table = T) + xlab("OS")
ggsurvplot(survfit(PFS ~ import$OLD_0_1, data = import), pval = T, risk.table = T) + xlab("PFS")


```

------------------------------------------------------------------------

### Sex - pval n.s.

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$SEX, data = import), pval = T, risk.table = T) + xlab("OS")
ggsurvplot(survfit(PFS ~ import$SEX, data = import), pval = T, risk.table = T) + xlab("PFS")

```

------------------------------------------------------------------------

### Hemoglobin \> 105 - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$HB_m_105, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$HB_m_105, data = import), pval = T, risk.table = T)+ xlab("PFS")


```

------------------------------------------------------------------------

### Platlets \> 150 \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$PLT_m_150, data = import), pval = T, risk.table = T) + xlab("OS")
ggsurvplot(survfit(PFS ~ import$PLT_m_150, data = import), pval = T, risk.table = T)+ xlab("PFS")
```

------------------------------------------------------------------------

### Plasma Cells \> 60 - pval n.s.

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$PC_M_60, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$PC_M_60, data = import), pval = T, risk.table = T)+ xlab("PFS")
```

------------------------------------------------------------------------

### LDH \> upper limit - pval n.s.

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$LDH_UL, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$LDH_UL, data = import), pval = T, risk.table = T)+ xlab("PFS")

```

------------------------------------------------------------------------

### I.S.S. staging - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$ISS, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$ISS, data = import), pval = T, risk.table = T)+ xlab("PFS")


```

------------------------------------------------------------------------

### Revised I.S.S. staging - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$R_ISS, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$R_ISS, data = import), pval = T, risk.table = T)+ xlab("PFS")

```

------------------------------------------------------------------------

### Ig Isotype - pval n.s.

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$IG_ISOTYPE_REV, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$IG_ISOTYPE_REV, data = import), pval = T, risk.table = T)+ xlab("PFS")
```

------------------------------------------------------------------------

### Light Chain Kappa/Lambda - pval n.s.

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$LIGHT_CHAIN, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$LIGHT_CHAIN, data = import), pval = T, risk.table = T)+ xlab("PFS")
```

------------------------------------------------------------------------

### ASCT Tx Yes/No - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$TX_I_LINE_1_0, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$TX_I_LINE_1_0, data = import), pval = T, risk.table = T)+ xlab("PFS")

```

------------------------------------------------------------------------

### Number of ASCT - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$NUMBERS_TX_FRONT_LINE, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$NUMBERS_TX_FRONT_LINE, data = import), pval = T, risk.table = T)+ xlab("PFS")

```

------------------------------------------------------------------------

### t(4;14) - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$FISH_T_4_14, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$FISH_T_4_14, data = import), pval = T, risk.table = T)+ xlab("PFS")

```

------------------------------------------------------------------------

### t(11;14) - pval n.s.

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$FISH_T_11_14, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$FISH_T_11_14, data = import), pval = T, risk.table = T)+ xlab("PFS")

```

------------------------------------------------------------------------

### t(14;16) - pval \*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$FISH_T_14_16, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$FISH_T_14_16, data = import), pval = T, risk.table = T)+ xlab("PFS")


```

------------------------------------------------------------------------

### t(14;20) - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$FISH_T_14_20, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$FISH_T_14_20, data = import), pval = T, risk.table = T)+ xlab("PFS")


```

------------------------------------------------------------------------

### t(6;14) - pval n.s.

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$FISH_T_6_14, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$FISH_T_6_14, data = import), pval = T, risk.table = T)+ xlab("PFS")


```

------------------------------------------------------------------------

### Hyperdiploidy (SNP) - pval \*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$HyperDiploidy, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$HyperDiploidy, data = import), pval = T, risk.table = T)+ xlab("PFS")

```

------------------------------------------------------------------------

### Del 17p Broad \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$`DEL_maj-broad_chr_17p`, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$`DEL_maj-broad_chr_17p`, data = import), pval = T, risk.table = T)+ xlab("PFS")

coxph(OS ~ import$`DEL_maj-broad_chr_17p` ,data = import) %>% summary
coxph(PFS ~ import$`DEL_maj-broad_chr_17p`, data = import) %>% summary
```

------------------------------------------------------------------------

### Del TP53 focal - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$`DEL_maj-focal_TP53`, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$`DEL_maj-focal_TP53`, data = import), pval = T, risk.table = T)+ xlab("PFS")

coxph(OS ~ import$`DEL_maj-focal_TP53`,data = import) %>% summary
coxph(PFS ~ import$`DEL_maj-focal_TP53`,data = import) %>% summary
```

------------------------------------------------------------------------

### Amp 1q Broad+Focal - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
# 1q MMrisk focal+broad 

ggsurvplot(survfit(OS ~ import$MMrisk_1q_all, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$MMrisk_1q_all, data = import), pval = T, risk.table = T)+ xlab("PFS")

coxph(OS ~ import$MMrisk_1q_all,data = import) %>% summary
coxph(PFS ~ import$MMrisk_1q_all,data = import) %>% summary
```

------------------------------------------------------------------------

### Del 13 Broad+Focal - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
# 13 MMrisk focal+broad
ggsurvplot(survfit(OS ~ import$MMrisk_13_all, data = import), pval = T, risk.table = T) + xlab("OS")
ggsurvplot(survfit(PFS ~ import$MMrisk_13_all, data = import), pval = T, risk.table = T) + xlab("PFS")

coxph(OS ~ import$MMrisk_13_all,data = import) %>% summary
coxph(PFS ~ import$MMrisk_13_all,data = import) %>% summary
```

------------------------------------------------------------------------

### MM MISK CLASSIFIER 1q&13 - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$MMrisk_CLASSIFIER_ALL, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$MMrisk_CLASSIFIER_ALL, data = import), pval = T, risk.table = T)+ xlab("PFS")

```

------------------------------------------------------------------------

### MMrisk class 1 - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}

ggsurvplot(survfit(OS ~ import$MMrisk1, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$MMrisk1, data = import), pval = T, risk.table = T)+ xlab("PFS")

coxph(OS ~ import$MMrisk1,data = import) %>% summary
coxph(PFS ~ import$MMrisk1,data = import) %>% summary
```

------------------------------------------------------------------------

### MMrisk class 2 - pval n.s.

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$MMrisk2, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$MMrisk2, data = import), pval = T, risk.table = T)+ xlab("PFS")
```

------------------------------------------------------------------------

### MMrisk class 3 - pval \*\*\*

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$MMrisk3, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$MMrisk3, data = import), pval = T, risk.table = T)+ xlab("PFS")

coxph(OS ~ import$MMrisk3,data = import) %>% summary
coxph(PFS ~ import$MMrisk3,data = import) %>% summary

```

------------------------------------------------------------------------

### MM RISK CLASSIFIER class 2 separated

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$MMrisk_AB_ALL, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$MMrisk_AB_ALL, data = import), pval = T, risk.table = T)+ xlab("PFS")

```

------------------------------------------------------------------------

### MMrisk class 1 VS Amp1q only VS other

```{r echo=FALSE, message=FALSE, warning=FALSE}
ggsurvplot(survfit(OS ~ import$risk1_VS_1q_VS_other, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$risk1_VS_1q_VS_other, data = import), pval = T, risk.table = T)+ xlab("PFS")

```

------------------------------------------------------------------------

### MMrisk class 1 VS Amp1q

```{r echo=FALSE, message=FALSE, warning=FALSE}
#____ Risk_1 vs only_1q _______
imp1q <- filter(import, `AMP_maj-broad_chr_1q` == 1)

OS1q <- Surv(imp1q$OS_MONTHS, imp1q$OS_EVENT)
PFS1q <- Surv(imp1q$PFS_I_MONTHS, imp1q$PFS_I_EVENT)

ggsurvplot(survfit(OS1q ~ imp1q$risk1_VS_1q_VS_other, data = imp1q), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS1q ~ imp1q$risk1_VS_1q_VS_other, data = imp1q), pval = T, risk.table = T)+ xlab("PFS")

coxph(OS1q ~ imp1q$risk1_VS_1q_VS_other,data = imp1q) %>% summary
coxph(PFS1q ~ imp1q$risk1_VS_1q_VS_other,data = imp1q) %>% summary
```

------------------------------------------------------------------------

### ULTRA MM RISK


```{r echo=FALSE, message=FALSE, warning=FALSE}


#______ ULTRA MMrisk ________

ggsurvplot(survfit(OS ~ import$ULTRA_MMRISK, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$ULTRA_MMRISK, data = import), pval = T, risk.table = T)+ xlab("PFS")

survfit(OS ~ import$ULTRA_MMRISK)
survfit(PFS ~ import$ULTRA_MMRISK)

#_____ ultra High _________

ggsurvplot(survfit(OS ~ import$Ultra_High, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$Ultra_High, data = import), pval = T, risk.table = T)+ xlab("PFS")

survfit(OS ~ import$Ultra_High)
survfit(PFS ~ import$Ultra_High)

coxph(OS ~ import$Ultra_High + strata(PROTOCOL_REV), data = import) %>% summary
coxph(PFS ~ import$Ultra_High + strata(PROTOCOL_REV), data = import) %>% summary


#_____ ultra Low _________

ggsurvplot(survfit(OS ~ import$Ultra_Low, data = import), pval = T, risk.table = T)+ xlab("OS")
ggsurvplot(survfit(PFS ~ import$Ultra_Low, data = import), pval = T, risk.table = T)+ xlab("PFS")

survfit(OS ~ import$Ultra_Low)
survfit(PFS ~ import$Ultra_Low)

coxph(OS ~ import$Ultra_Low + strata(PROTOCOL_REV), data = import) %>% summary
coxph(PFS ~ import$Ultra_Low + strata(PROTOCOL_REV), data = import) %>% summary




```



------------------------------------------------------------------------

------------------------------------------------------------------------

## MULTIVARIATE ANALYSIS

### Model GENOMIC (MM risk class 1, del TP53, t(4;14), t(14;16), t(14;20), del 1p, Hyperdiploidy. Stratified per Protocol)

```{r echo=FALSE, message=FALSE, warning=FALSE}
coxph(OS ~ strata(PROTOCOL_REV) + MMrisk1 + `DEL_maj-focal_TP53` + FISH_T_4_14 + `DEL_maj-broad_chr_1p` + HyperDiploidy + FISH_T_14_16 + FISH_T_14_20 + HyperDiploidy, data = import) %>% summary

coxph(PFS ~ strata(PROTOCOL_REV) + MMrisk1 + `DEL_maj-focal_TP53` + FISH_T_4_14 + `DEL_maj-broad_chr_1p` + HyperDiploidy + FISH_T_14_16 + FISH_T_14_20 + HyperDiploidy, data = import) %>% summary

```

### Model GENOMIC & CLINIC (Old age, HB\>105, PLT\>150, TX first line, Hyperdiploidy, MM risk class 1, del TP53, t(4;14), t(14;16), t(14;20), del 1p. Stratified per Protocol)

```{r echo=FALSE, message=FALSE, warning=FALSE}
coxph(OS ~ strata(PROTOCOL_REV) + MMrisk1 + `DEL_maj-focal_TP53` + FISH_T_4_14 + `DEL_maj-broad_chr_1p` + OLD_0_1 + HB_m_105 + PLT_m_150 + HyperDiploidy + FISH_T_14_16 + FISH_T_14_20 + TX_I_LINE_1_0, data = import) %>% summary


coxph(PFS ~ strata(PROTOCOL_REV) + MMrisk1 + `DEL_maj-focal_TP53` + FISH_T_4_14 + `DEL_maj-broad_chr_1p` + OLD_0_1 + HB_m_105 + PLT_m_150 + HyperDiploidy + FISH_T_14_16 + FISH_T_14_20 + TX_I_LINE_1_0, data = import) %>% summary

```



### ULTRA - Model GENOMIC & CLINIC (Old age, HB\>105, PLT\>150, TX first line, Hyperdiploidy, ULTRA High Risk, del TP53, t(4;14), t(14;16), t(14;20), del 1p. Stratified per Protocol)

```{r echo=FALSE, message=FALSE, warning=FALSE}
coxph(OS ~ strata(PROTOCOL_REV) + Ultra_High + `DEL_maj-focal_TP53` + FISH_T_4_14 + `DEL_maj-broad_chr_1p` + OLD_0_1 + HB_m_105 + PLT_m_150 + HyperDiploidy + FISH_T_14_16 + FISH_T_14_20 + TX_I_LINE_1_0, data = import) %>% summary


coxph(PFS ~ strata(PROTOCOL_REV) + Ultra_High + `DEL_maj-focal_TP53` + FISH_T_4_14 + `DEL_maj-broad_chr_1p` + OLD_0_1 + HB_m_105 + PLT_m_150 + HyperDiploidy + FISH_T_14_16 + FISH_T_14_20 + TX_I_LINE_1_0, data = import) %>% summary

```


